docs/content/FormControl.mdx

```diff
@@ -0,0 +1,429 @@
+---
+componentId: formControl
+title: FormControl
+status: Alpha
+description: The FormControl component is used to render a labelled text input and, optionally, associated validation text and hint text.
+source: https://github.com/primer/react/blob/main/src/FormControl/FormControl.tsx
+storybook: '/react/storybook?path=/story/forms-inputfield--text-input-field'
+---
+
+import {FormControl, TextInputWithTokens, Autocomplete, Select, Textarea, Checkbox, Radio} from '@primer/react'
+import {MarkGithubIcon} from '@primer/octicons-react'
+
+## Examples
+
+### Basic
+
+```jsx live
+<FormControl>
+  <FormControl.Label>Name</FormControl.Label>
+  <TextInput />
+</FormControl>
+```
+
+### `variant="stack"` (default) inputs
+
+```javascript live noinline
+const DifferentInputs = () => {
+  const [tokens, setTokens] = React.useState([
+    {text: 'zero', id: 0},
+    {text: 'one', id: 1},
+    {text: 'two', id: 2}
+  ])
+  const onTokenRemove = tokenId => {
+    setTokens(tokens.filter(token => token.id !== tokenId))
+  }
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      sx={{
+        '> * + *': {
+          marginTop: 4
+        }
+      }}
+    >
+      <FormControl>
+        <FormControl.Label>TextInputWithTokens</FormControl.Label>
+        <TextInputWithTokens onTokenRemove={onTokenRemove} tokens={tokens} />
+      </FormControl>
+      <FormControl>
+        <FormControl.Label>Autocomplete</FormControl.Label>
+        <Autocomplete>
+          <Autocomplete.Input block />
+          <Autocomplete.Overlay>
+            <Autocomplete.Menu
+              items={[
+                {text: 'css', id: 0},
+                {text: 'css-in-js', id: 1},
+                {text: 'styled-system', id: 2},
+                {text: 'javascript', id: 3},
+                {text: 'typescript', id: 4},
+                {text: 'react', id: 5},
+                {text: 'design-systems', id: 6}
+              ]}
+              selectedItemIds={[]}
+            />
+          </Autocomplete.Overlay>
+        </Autocomplete>
+      </FormControl>
+      <FormControl>
+        <FormControl.Label>Select</FormControl.Label>
+        <Select>
```

Comment:
Think there may be an issue with `Select` here, as the dropdown isn't triggered unless you press the text. Even pressing arrows doesn't work.

https://user-images.githubusercontent.com/13340707/152513393-00d6afd2-5173-48bd-9cce-c42d9f7d5803.mov